src/Workspaces/VisualBasic/Portable/LanguageServices/VisualBasicSyntaxFactsService.vb

```diff
@@ -662,7 +662,17 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                     Return Nothing
                 End If
 
-                Return TextSpan.FromBounds(method.BlockStatement.Span.End, method.EndBlockStatement.SpanStart)
+                ' We don't want to include the BlockStatement or any trailing trivia up to and including its statement
+                ' terminator in the span. Instead, we use the start of the first statement's leading trivia (if any) up
+                ' to the start of the EndBlockStatement. If there aren't any statements in the block, we use the start
+                ' of the EndBlockStatements leading trivia.
+
+                Dim firstStatement = method.Statements.FirstOrDefault()
+                Dim spanStart = If(firstStatement IsNot Nothing,
+                                   firstStatement.FullSpan.Start,
+                                   method.EndBlockStatement.FullSpan.Start)
+
+                Return TextSpan.FromBounds(spanStart, method.EndBlockStatement.SpanStart)
```

Comment:
Sure. I'll take a stab at writing something up. The methods for getting a speculative semantic model seem to be mostly mysterious.